### ISSUE TITLE
Batch is broken.

### ISSUE THREAD

**Issue by deoxen0n2** [algolia/algoliasearch-client-node#27]
I just saw this commit: https://github.com/algolia/algoliasearch-client-node/commit/69b09fa6e093bc4eac753e1f0ef132148d6273cf, `_batch` is called with 4 parameters.

But in the current implementation:
https://github.com/algolia/algoliasearch-client-node/blob/V1/src/algoliasearch-node.js#L1350

```js
_batch: function(objects, action, callback) {
  var postObj = {requests:[]};
  for (var i = 0; i < objects.length; ++i) {
    var request = { action: action,
    body: objects[i] };
    if (!_.isUndefined(objects[i].objectID)) {
      request.objectID = objects[i].objectID;
    }
    postObj.requests.push(request);
  }
  this.as._request('POST', '/1/indexes/' + encodeURIComponent(this.indexName) + '/batch', postObj, 'write', callback);
},
```

Batch is accepting just 3 parameters. This gives me `TypeError: string is not a function` when I try to call the function that use `_batch`. Are you guys are on the ongoing of new implementation?

Sorry for my English :).

**Comment by vvo** [comment]
good catch @deoxen0n2, going to publish a patch

**Comment by vvo** [comment]
Ok, landed as 1.8.3 @deoxen0n2
